lib/trackler/problem.rb

```diff
@@ -1,20 +1,28 @@
 require 'yaml'
+require_relative 'metadata'
+require_relative 'description'
+require_relative 'null_track'
 
 module Trackler
   # Problem is a language-independent definition of an exercise.
   class Problem
-    attr_reader :slug, :root
-    def initialize(slug, root)
+    attr_reader :slug, :root, :metadata
+    def initialize(slug, root, track = NullTrack.new)
       @slug = slug
       @root = root
+      @file_root = File.join(root, 'common', 'exercises', self.slug)
+      @repo_root = "https://github.com/exercism/x-common/blob/master/exercises/%s/" % self.slug
+
+      @metadata = Metadata.for(problem: self, track: track)
+      self.description_object = Description.for(problem: self, track: track)
```

Comment:
I see why we have to name it something like this, and I'm wishing (out loud) that I could think of a better name. But so far I haven't come up with anything.